consensus/polybft/fsm.go

```diff
@@ -535,15 +541,23 @@ func (f *fsm) Insert(p *pbft.SealedProposal) error {
 	// Write extar data to header
 	f.block.Block.Header.ExtraData = append(make([]byte, 32), extra.MarshalRLPTo(nil)...)
 
-	if err := f.backend.CommitBlock(f.block); err != nil {
+	// commit exit events only when we finalize a block
+	events, err := getExitEventsFromReceipts(f.epochNumber, f.block.Block.Number(), f.blockBuilder.Receipts())
+	if err != nil {
 		return err
 	}
 
-	if err := f.postInsertHook(); err != nil {
+	if len(events) > 0 {
+		if err := f.checkpointBackend.InsertExitEvents(events); err != nil {
+			return err
+		}
+	}
```

Comment:
Should we move this part after `f.backend.CommitBlock(f.block)` or even to the `postInsertHook` perhaps? 
Because at that point we are sure that block is written to the local storage (i.e. finalized).

Reply:
Yes, but what happens if we do not insert these events for some reason? An error occurs. Should we move to another round and try again? To me that is a correct way to have a consistent state, but lets discuss it further on sync.

Reply:
> Should we move to another round and try again?
Exactly. If `Insert` function fails for any reason, we are going to proceed to a new round where we would reset our internal state.
Although I don't see that we are re-injecting transactions back to the tx pool if `Insert` fails. That could be a problem, because in that case we would lose such transactions (with exit event logs).

Reply:
Ok, so agreement here is to go as it is. The issue with not re-injecting transactions back to the tx pool if finalization of block fails will be addressed through separate issue, since it's not really an issue on checkpointing, but an existing one.